src/cargo/core/compiler/job_queue.rs

```diff
@@ -221,9 +224,18 @@ impl<'a> JobQueue<'a> {
             // jobserver interface is architected we may acquire a token that we
             // don't actually use, and if this happens just relinquish it back
             // to the jobserver itself.
-            tokens.truncate(self.active - 1);
-
-            match self.rx.recv().unwrap() {
+            tokens.truncate(self.active.len() - 1);
+
+            let count = queue_len - self.queue.len();
```

Comment:
Micronit: `queue.len() - queueue_len` looks funny. Perhaps, rename it to `total -self.que.len()`?